src/libraries/System.Private.CoreLib/src/System/Resources/ManifestBasedResourceGroveler.cs

```diff
@@ -34,6 +36,11 @@ namespace System.Resources
     internal partial class ManifestBasedResourceGroveler : IResourceGroveler
     {
         private readonly ResourceManager.ResourceManagerMediator _mediator;
+        private static readonly bool s_allowUsingReflectionToLoadTypes = AppContext.TryGetSwitch("System.Resources.AllowReflectionForNonPrimitiveObjects", out bool allowReflection) ? allowReflection : true;
+
+        // Adding static method that is not inlined so it can be substituted away by the linker to disallow the use of serialization.
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static bool AllowUsingReflectionToLoadTypes() => s_allowUsingReflectionToLoadTypes;
```

Comment:
More compact way to write this:
```
private static bool AllowUsingReflectionToLoadTypes() =>
      AppContext.TryGetSwitch("System.Resources.AllowReflectionForNonPrimitiveObjects", out bool allowReflection) ? allowReflection : true;`
``

Reply:
Wouldn't this will try to re-read the switch value every time it gets called? I mean will not use a cached value?

Reply:
You are right ... I meant to say:
```
private static bool AllowUsingReflectionToLoadTypes { get; } =
      AppContext.TryGetSwitch("System.Resources.AllowReflectionForNonPrimitiveObjects", out bool allowReflection) ? allowReflection : true;
```